src-vue/src/utils/thunderstore/version.ts

```diff
@@ -0,0 +1,32 @@
+import {ThunderstoreMod} from "../../../../src-tauri/bindings/ThunderstoreMod";
+import {NorthstarMod} from "../../../../src-tauri/bindings/NorthstarMod";
+import {store} from "../../plugins/store";
+
+/**
+ * Strips off a Thunderstore dependency string from its version
+ * (e.g. "taskinoz-WallrunningTitans-1.0.0" to
+ * "taskinoz-WallrunningTitans").
+ **/
+function getThunderstoreDependencyStringPrefix(dependency: string): string {
+    const dependencyStringMembers = dependency.split('-');
+    return `${dependencyStringMembers[0]}-${dependencyStringMembers[1]}`;
+}
+
+function isThunderstoreModOutdated(mod: ThunderstoreMod): boolean {
+    // Ensure mod is up-to-date.
+    const tsModPrefix = getThunderstoreDependencyStringPrefix(mod.versions[0].full_name);
+    const matchingMods: NorthstarMod[] = store.state.installed_mods.filter((mod: NorthstarMod) => {
+        if (!mod.thunderstore_mod_string) return false;
+        return getThunderstoreDependencyStringPrefix(mod.thunderstore_mod_string!) === tsModPrefix;
```

Comment:
What does the trailing exclamation mark do?

Reply:
It ensures that `thunderstore_mod_string` is not null

Reply:
Ah, it's a TypeScript thing to tell the compiler that value will never be `null` here, right?

Reply:
yep!